setup_mws3.py

```diff
@@ -0,0 +1,168 @@
+import subprocess
+
+def subprocess_cmd(command):
+    """
+        Function used to run shell commands
+        This is Python that creates a Python env
+        which then runs Python commands - hhmmm
+    """
+    process = subprocess.Popen(command,stdout=subprocess.PIPE, shell=True)
+    proc_stdout = process.communicate()[0].strip()
+    print proc_stdout
+
+
+
+import MySQLdb as mdb
+import sys
+import os
+
+filedir_path = os.path.dirname(os.path.realpath(__file__))
+
+# NOTE: A user could provide mangled strings to execute commands as the root mysql user
+#   - but they have the root password anyway!!
+
+#Test for root password
+rootpass = raw_input("Root password for database:")
+
+try:
+
+    con = mdb.connect('localhost', 'root', rootpass, '');
+
+except mdb.Error, e:
+    print "Error - please check root database password"
+    sys.exit(1)
+
+finally:
+    try:
+        if con:
+            cur = con.cursor()
+            cur.execute("SELECT VERSION()")
+            ver = cur.fetchone()
+            print "Database version : %s " % ver
+
+            con.close()
+    except NameError:
+        print " --Try again?-- "
+
+
+
+print "The following settings can be edited later in the settings_local.py file:"
+dbname=""
+while (len(dbname) < 1 or len(dbname) > 9 ):
+    dbname = raw_input("Please provide a name for your database: (max length: 9)")
+dbname = "pc_%s" % dbname
+password = raw_input("Enter a password your applicaton will use to access this database: ")
+secret_key = raw_input("Enter a secret key for your application (this will be used by the application): ")
+development = raw_input("Do you wish to load example data? [N]y: ")
+is_dev=False
+if development.lower() == "y":
+    is_dev=True
+
+username="%s_web" % dbname
+
+
+
+try:
+    con = mdb.connect('localhost', 'root', rootpass, 'test');
+    cur = con.cursor()
+
+    cur.execute("create database %s " % dbname)
+    cur.execute("create user %s@'localhost' identified by '%s'" % (username,password) )
+    cur.execute("grant all on %s.* to %s@'localhost'" % (dbname,username) )
+
+
+except mdb.Error, e:
+
+    print "****\nError we can not proceed:\n\n %d: %s \n****" % (e.args[0],e.args[1])
+    sys.exit(1)
+
+finally:
+
+    #return the config strings
+
+    f = open("%s/edpcmentoring/edpcmentoring/settings_local.py" % filedir_path, 'w')
```

Comment:
It's more "Pythonic" to do something like:

```python
with open('....') as f:
    f.write(foo)
```

This has the added advantage of not needing the ``f.close()`` call.

I think on Python 2.6, one gets this via a ``from __future__ import with_statement`` at the top of the file.